src/components/page/page_header/page_header_content.tsx

```diff
@@ -344,25 +344,34 @@ export const EuiPageHeaderContent: FunctionComponent<EuiPageHeaderContentProps>
 
   let rightSideFlexItem;
   if (rightSideItems && rightSideItems.length) {
-    const wrapWithFlex = () => {
-      return rightSideItems.map((item, index) => {
-        return (
-          <EuiFlexItem grow={false} key={index}>
-            {item}
-          </EuiFlexItem>
-        );
-      });
+    const wrapWithFlex = (reverse: boolean) => {
+      const reverseRightSideItems = reverse
+        ? rightSideItems.reverse()
+        : undefined;
+
+      if (reverseRightSideItems) {
+        return reverseRightSideItems.map((item, index) => {
+          return (
+            <EuiFlexItem grow={false} key={index}>
+              {item}
+            </EuiFlexItem>
+          );
+        });
+      } else {
+        return rightSideItems.map((item, index) => {
+          return (
+            <EuiFlexItem grow={false} key={index}>
+              {item}
+            </EuiFlexItem>
+          );
+        });
+      }
```

Comment:
```suggestion
    const wrapWithFlex = (reverse: boolean) => {
      if (reverse) {
        return reverseRightSideItems.reverse().map((item, index) => {
          return (
            <EuiFlexItem grow={false} key={index}>
              {item}
            </EuiFlexItem>
          );
        });
      } else {
        return rightSideItems.map((item, index) => {
          return (
            <EuiFlexItem grow={false} key={index}>
              {item}
            </EuiFlexItem>
          );
        });
      }
```

Reply:
I agree with Trevor, but I think you're both writing one `.map` too many :)

```suggestion
    const wrapWithFlex = (reverse: boolean) => {
      const itemsToRender = reverse
        ? [...rightSideItems].reverse()
        : rightSideItems;

      return itemsToRender.map((item, index) => (
        <EuiFlexItem grow={false} key={index}>
          {item}
        </EuiFlexItem>
      ));
```